pkg/server/server.go

```diff
@@ -0,0 +1,142 @@
+package server
+
+import (
+	"bufio"
+	"encoding/binary"
+	"encoding/json"
+	"io"
+	"net"
+	"time"
+
+	"github.com/wavesplatform/gowaves/pkg/proto"
+	"go.uber.org/zap"
+)
+
+type Server struct {
+	BootPeerAddrs []string
+	Listen        string
+}
+
+func handleRequest(conn net.Conn) {
+}
+
+func handleClient(conn net.Conn) {
+	handshake := proto.Handshake{Name: "wavesT",
+		VersionMajor:      0x0,
+		VersionMinor:      0xe,
+		VersionPatch:      0x4,
```

Comment:
Actually, fork detector should try different versions during handshake. If it receives handshake it should reply with received version.